ps_socialfollow.php

```diff
@@ -187,11 +202,32 @@ public function getConfigFieldsValues()
     {
         $result = [];
         foreach (static::SOCIAL_NETWORKS as $social) {
-            $result['blocksocial_' . $social] = Configuration::get('BLOCKSOCIAL_' . strtoupper($social));
+            if (!empty(Configuration::get('BLOCKSOCIAL_' . strtoupper($social)))) {
+                $this->upgradeConfiguration('BLOCKSOCIAL_' . strtoupper($social));
+            }
+            foreach ($this->context->controller->getLanguages() as $lang) {
+                $id_lang = $lang['id_lang'];
+                $conf = Configuration::get('BLOCKSOCIAL_' . strtoupper($social), $id_lang);
+                $result['blocksocial_' . $social][$id_lang] = $conf;
+            }
         }
         return $result;
     }
 
+    protected function upgradeConfiguration($name)
+    {
+        $conf = Configuration::get($name);
+        if (!empty($conf) && !is_array($conf)) {
+            $conf_localized = [];
```

Comment:
Nice-to-have: Maybe we can use private methods to help reader understand this statement that is not easy to read ?

```suggestion
            $localizedConfiguration = $this->buildLocalizedConfiguration($currentConfiguration, $enabledLanguages)
```
```

Reply:
I disagree. This is used only once. And it is in a function that is run only once in the lifetime of the module. The problem is what you pointed out before: The function needs documentation. This is essential here.